In [1]:
import numpy as np
import pandas as pd

In [2]:
gammadist = [1.691, 1.437, 8.221, 5.976,
1.116, 4.435, 2.345, 1.782,
3.810, 4.589, 5.313, 10.90,
2.649, 2.432, 1.581, 2.432,
1.843, 2.466, 2.833, 2.361]

In [3]:
def M(arr):
    hoha = 0
    for i in range(len(arr)):
        hoha += np.log(arr[i])
    return np.log(np.mean(arr))  + hoha

In [4]:
M_f = M(gammadist)
M_f

22.611700175083392

In [5]:
1/M_f

0.04422489208051389

Didapat nilai 1/M = 0.04422489208051389, sehingga didapat nilai $\beta$

## EAS 2020 NO 3

Metode Additive<br>
$Z_{i+1} = (Z_i + \mu) mod P$<br>
ada 2 parameter, $\mu$ dan P, tentuin sendiri $Z_0$, biar bagus, ambil $Z_0$ dan $\mu$ Prima, dan P prima yang besar

In [6]:
Zo = 3
listOfRandomNumber = [Zo]
def additiveMethod(mu, p, length):
    for i in range(1, length):
        rand = (listOfRandomNumber[i-1] + mu) % p
        listOfRandomNumber.append(rand)
    return listOfRandomNumber

In [12]:
t = additiveMethod(11, 1039, 1000)
t

[3,
 14,
 25,
 36,
 47,
 58,
 69,
 80,
 91,
 102,
 113,
 124,
 135,
 146,
 157,
 168,
 179,
 190,
 201,
 212,
 223,
 234,
 245,
 256,
 267,
 278,
 289,
 300,
 311,
 322,
 333,
 344,
 355,
 366,
 377,
 388,
 399,
 410,
 421,
 432,
 443,
 454,
 465,
 476,
 487,
 498,
 509,
 520,
 531,
 542,
 553,
 564,
 575,
 586,
 597,
 608,
 619,
 630,
 641,
 652,
 663,
 674,
 685,
 696,
 707,
 718,
 729,
 740,
 751,
 762,
 773,
 784,
 795,
 806,
 817,
 828,
 839,
 850,
 861,
 872,
 883,
 894,
 905,
 916,
 927,
 938,
 949,
 960,
 971,
 982,
 993,
 1004,
 1015,
 1026,
 1037,
 9,
 20,
 31,
 42,
 53,
 64,
 75,
 86,
 97,
 108,
 119,
 130,
 141,
 152,
 163,
 174,
 185,
 196,
 207,
 218,
 229,
 240,
 251,
 262,
 273,
 284,
 295,
 306,
 317,
 328,
 339,
 350,
 361,
 372,
 383,
 394,
 405,
 416,
 427,
 438,
 449,
 460,
 471,
 482,
 493,
 504,
 515,
 526,
 537,
 548,
 559,
 570,
 581,
 592,
 603,
 614,
 625,
 636,
 647,
 658,
 669,
 680,
 691,
 702,
 713,
 724,
 735,
 746,
 757,
 768,
 779,
 790,
 801,
 812,
 8

Metode Inversi<br>
$x_i = \sqrt{2\cdot u_i}$

In [8]:
import math

In [9]:
def inversionMethod():
    inversionList = []
    for i in range(len(listOfRandomNumber)):
        inversionList.append(math.sqrt(2*listOfRandomNumber[i]))
    return inversionList

In [11]:
inversionMethod()

[2.449489742783178,
 5.291502622129181,
 7.0710678118654755,
 8.48528137423857,
 9.695359714832659,
 10.770329614269007,
 11.74734012447073,
 12.649110640673518,
 13.490737563232042,
 14.2828568570857,
 15.033296378372908,
 15.748015748023622,
 16.431676725154983,
 17.08800749063506,
 17.72004514666935,
 18.33030277982336,
 18.920887928424502,
 19.493588689617926,
 20.049937655763422,
 20.591260281974,
 21.118712081942874,
 21.633307652783937,
 22.135943621178654,
 22.627416997969522,
 23.108440016582687,
 23.57965224510319,
 24.041630560342615,
 24.49489742783178,
 24.939927826679853,
 25.37715508089904,
 25.80697580112788,
 26.229754097208,
 26.645825188948457,
 27.055498516937366,
 27.459060435491963,
 27.85677655436824,
 28.24889378365107,
 28.635642126552707,
 29.017236257093817,
 29.393876913398138,
 29.765752132274432,
 30.133038346638727,
 30.495901363953813,
 30.854497241083024,
 31.20897306865447,
 31.559467676119,
 31.906112267087632,
 32.2490309931942,
 32.58834147360065,
 